### PR TITLE
[CI] Add GitHub Actions for Windows on self-hosted runners

### PR DESCRIPTION
**Pull request by alexbatashev**
Update configs to run on self-hosted runners.

### CODE REVIEW

**Comment by bader** [comment]
@alexbatashev, please, disable linux precommit for this PR.

File: .github/workflows/sycl_precommit.yml [hunk 1]
```diff
@@ -40,3 +40,9 @@ jobs:
       build_artifact_suffix: "default"
       build_cache_suffix: "default"
       lts_config: "hip_amdgpu;ocl_x64;ocl_gen9;l0_gen9"
+
+  windows_default:
+    name: Window
+    needs: lint
+    if: always() && (success() || contains(github.event.pull_request.labels.*.name, 'ignore-lint'))
```
Comment (bader) [inline .github/workflows/sycl_precommit.yml:47]: @alexbatashev, are we ready to enable it in pre-commit?
I have concerns about that.
Can we enable nightly and post-commit validation on Windows first and bring pre-commit later?
I'd like to make sure that these checks do not introduce false positives and we have enough resources to handle pre-commit requests.

File: .github/workflows/sycl_windows_build_and_test.yml [hunk 2]
```diff
@@ -2,54 +2,80 @@ name: SYCL Windows Test
...
-      with:
-        path: src
-        fetch-depth: 1
-    - name: Install dependencies
-      shell: cmd
-      run: |
-        choco install -y ninja
-        choco install -y sccache --version 0.2.15
-        refreshenv
     - uses: ilammy/msvc-dev-cmd@9f8ae839b01883414208f29e3e24524387f48e1f
```
Comment (bader) [inline .github/workflows/sycl_windows_build_and_test.yml:17]: Why SHA instead of a release version?

Comment (alexbatashev) [reply .github/workflows/sycl_windows_build_and_test.yml:17]: Tags can change, commit SHA will always point to the same version of code.

File: .github/workflows/sycl_precommit.yml (hunk 1 above)
Comment (alexbatashev) [reply .github/workflows/sycl_precommit.yml]: done

**Comment by alexbatashev** [comment]
failures with new Pass Manager are expected

**Review by bader — APPROVED** [review]

**Pull request merged**